ocaml/xapi/server_helpers.ml

```diff
@@ -58,7 +67,7 @@ let parameter_count_mismatch_failure func expected received =
 
 (** WARNING: the context is destroyed when execution is finished if the task is not forwarded, in database and not called asynchronous. *)
 (*  FIXME: This function should not be used for external call : we should add a proper .mli file to hide it. *)
-let exec_with_context ~__context ?marshaller ?f_forward ?(called_async=false) ?(has_task=false) f =
+let exec_with_context ~__context ~need_complete ?marshaller ?f_forward ?(called_async=false) f =
   (* Execute fn f in specified __context, marshalling result with "marshaller".
      If has_task is set then __context has a real task in it that has to be completed. *)
```

Comment:
`has_task` is gone.

Reply:
yes,

we demand the caller compute `need_complete` instead, as that is all that `has_task` was used for

Reply:
Yep, so that code comment needs to go away.

Reply:
ah - done